pkg/controller/config/config_controller_test.go

```diff
@@ -664,3 +666,61 @@ func unstructuredFor(gvk schema.GroupVersionKind, name string) *unstructured.Uns
 type testExpectations interface {
 	IsExpecting(gvk schema.GroupVersionKind, nsName types.NamespacedName) bool
 }
+
+func ensureDeleted(ctx context.Context, c client.Client, toDelete client.Object) func() error {
+	return func() error {
+		instance, ok := toDelete.DeepCopyObject().(client.Object)
+		if !ok {
+			return fmt.Errorf("instance was %T which is not a client.Object", instance)
+		}
+
+		key := client.ObjectKey{Namespace: instance.GetNamespace(), Name: instance.GetName()}
+		gvk := instance.GetObjectKind().GroupVersionKind()
+
+		u := &unstructured.Unstructured{}
+		u.SetGroupVersionKind(gvk)
+
+		err := c.Get(ctx, key, u)
+		if apierrors.IsNotFound(err) {
+			return nil
+		} else if err != nil {
+			return err
+		}
+
+		if !u.GetDeletionTimestamp().IsZero() {
+			return fmt.Errorf("waiting for deletion: %v %v", gvk, key)
+		}
+
+		err = c.Delete(ctx, instance)
```

Comment:
nit: we could set Namespace and Name on `u` to reference the original resource, and avoid the deepcopy.

Reply:
Good point.